util/fsdiffer/beforeafter.go

```diff
@@ -0,0 +1,63 @@
+package fsdiffer
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// The BeforeAfterFSDiffer is used to generate changes in a given directory
+// between two different points in time.
+type BeforeAfterFSDiffer struct {
+	dir    string
+	before map[string]fileInfo
+}
+
+// NewBeforeAfterFSDiffer creates a new BeforeAfterFSDiffer that will report
+// changes on the given directory.
+func NewBeforeAfterFSDiffer(dir string) *BeforeAfterFSDiffer {
+	return &BeforeAfterFSDiffer{dir: dir}
+}
+
+// Start will walk the provided directory, and record its state for use as the
+// before part of the changes returned by Diff.
+func (ba *BeforeAfterFSDiffer) Start() error {
```

Comment:
I'm struggling to think of a use case for initialising a BeforeAfterFSDiffer and not immediately calling Start() - what about just collapsing this into the constructor?